sway/input/seat.c

```diff
@@ -73,6 +73,24 @@ static void seat_node_destroy(struct sway_seat_node *seat_node) {
 	free(seat_node);
 }
 
+void seat_notify_key_press(struct sway_seat *seat) {
+	if (seat->cursor->hidden) {
+		return;
+	}
+
+	const struct seat_config *seat_config = seat_get_config(seat);
+	if (!seat_config) {
+		seat_config = seat_get_config_by_name("*");
+		if (!seat_config) {
+			return;
+		}
+	}
+
+	if (seat_config->hide_cursor_when_typing == true) {
```

Comment:
Don't explicitly compare with `true`.

Reply:
The thing is, `hide_cursor_when_typing` actually has three possible value: `true`, `false` and `-1` which stands for default/unset (every field in `seat_config` supports this in order to support merging configs). Therefore, if I remove the explicit comparison with `true` it means that a value of `-1` will also pass the `if` check, which is something I don't want.
I was thinking about making this an enum, but I saw that `fallback` is another field in `seat_config` which is kind of a boolean and basically has these three possible states, so I kinda copied from it.

Perhaps it would make sense to add a comment here explaining why this is an explicit comparison with `true`? Or maybe still use an enum even though it would be different from the `fallback` field? What do you think?

Reply:
The thing is, it actually has three possible values: `true`, `false`, and `-1` (default). Now that I'm looking at the code again I see that some others fields in `seat_config` that have similar three values use enums, so I'll use an enum here as well.